common/tests/unittests/test_memory_limit.py

```diff
@@ -0,0 +1,21 @@
+def test_memory_mocking():
+    import psutil
+    from autogluon.common.utils.resource_utils import ResourceManager
+
+    assert psutil.virtual_memory().total == ResourceManager._get_memory_size()
+    assert psutil.virtual_memory().available == ResourceManager._get_available_virtual_mem()
```

Comment:
I imagine this unit test could randomly fail if total/available memory changes between psutil calls.

Reply:
Proposing a fix based on a moneky patch in updated code.